### PR TITLE
This implements the ability to save snapshots of the aggregation source data 

### PR DESCRIPTION
**Pull request by kristianschneider**

### CODE REVIEW

File: source/databricks/geh_stream/aggregation_utils/services/aggregation_postprocessor.py [hunk 1]
```diff
@@ -16,18 +16,32 @@
...
+        result_base = "Results"
 
-    for key, value in results.items():
-        path = "{0}/{1}/{2}.json.gz".format(result_path, nowstring, key)
-        blob_service.upload_blob(value, path)
-        coordinator_service.notify_coordinator(path)
+        for key, value in results.items():
+            path = "{0}/{1}/{2}".format(result_base, now_path_string, key)
+            result_path = "abfss://{0}@{1}.dfs.core.windows.net/{2}".format(args.input_storage_container_name, args.input_storage_account_name, path)
+            stringFormatedTimeDf = value.withColumn("time_start", date_format(col("time_window.start"), "yyyy-MM-dd'T'HH:mm:ss'Z'")) \
```
Comment (mknic) [inline source/databricks/geh_stream/aggregation_utils/services/aggregation_postprocessor.py:35]: I'm not sure what is up and down in the way of handling dates around the system, but I really don't like this hardcoded date format string. Wouldn't it be better to locate it somewhere else where all modules can reach it?

**Review by mknic — CHANGES_REQUESTED** [review]
Looks good. One very minor suggested change and a question in regards to date format handling.

File: source/databricks/geh_stream/aggregation_utils/aggregators/aggregate_quality.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 from pyspark.sql import DataFrame
-from pyspark.sql.functions import col, when, window, count, year, month, dayofmonth, hour
+from pyspark.sql.functions import col, when, window, count, year, month, dayofmonth, hour,lit
```
[the remark this answers is not in the document]
Comment (mknic) [reply source/databricks/geh_stream/aggregation_utils/aggregators/aggregate_quality.py]: Oh1 The `lit` can actually be removed. It looks like it's not used anymore.

File: source/coordinator/GreenEnergyHub.Aggregation.Coordinator/CoordinationTriggers.cs [hunk 3]
```diff
@@ -133,6 +124,57 @@ public CoordinationTriggers(ICoordinatorService coordinatorService)
...
+                throw new ArgumentNullException(nameof(req));
+            }
+
+            try
+            {
+                // Handle gzip replies
+                var decompressedReqBody = await DecompressedReqBody(req);
+
+                var resultId = req.Headers["result-id"].First();
+                var snapshotPath = req.Headers["process-type"].First();
```
Comment (ZavezX) [inline source/coordinator/GreenEnergyHub.Aggregation.Coordinator/CoordinationTriggers.cs:146]: Should `snapshotPath` not map to another value?

File: source/databricks/aggregation-jobs/aggregation_trigger.py [hunk 4]
```diff
@@ -13,13 +13,12 @@
 # limitations under the License.
 
 # Uncomment the lines below to include modules distributed by wheel
-# import sys
-# sys.path.append(r'/workspaces/geh-aggregations/source/databricks')
+import sys
+sys.path.append(r'/workspaces/geh-aggregations/source/databricks')
```
Comment (ZavezX) [inline source/databricks/aggregation-jobs/aggregation_trigger.py:17]:
```suggestion
#import sys
#sys.path.append(r'/workspaces/geh-aggregations/source/databricks')
```

File: source/databricks/geh_stream/aggregation_utils/aggregators/aggregate_quality.py [hunk 5]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 from pyspark.sql import DataFrame
-from pyspark.sql.functions import col, when, window, count, year, month, dayofmonth, hour
+from pyspark.sql.functions import col, when, window, count, year, month, dayofmonth, hour, lit
```
Comment (ZavezX) [inline source/databricks/geh_stream/aggregation_utils/aggregators/aggregate_quality.py:16]:
```suggestion
from pyspark.sql.functions import col, when, window, count, year, month, dayofmonth, hour
```

File: source/databricks/geh_stream/aggregation_utils/services/aggregation_postprocessor.py (hunk 1 above)
Comment (kristianschneider) [reply source/databricks/geh_stream/aggregation_utils/services/aggregation_postprocessor.py:35]: We'll keep it until further

File: source/coordinator/GreenEnergyHub.Aggregation.Coordinator/CoordinationTriggers.cs (hunk 3 above)
Comment (kristianschneider) [reply source/coordinator/GreenEnergyHub.Aggregation.Coordinator/CoordinationTriggers.cs]: YES!

**Review by mknic — APPROVED** [review]

**Review by johevemi — APPROVED** [review]

**Pull request merged**